src/typesAndDefaults.ts

```diff
@@ -54,17 +54,24 @@ export interface OptionsInterface<C extends Context, RT extends RedisType> {
 
   /**
    * @param ctx Is the context object you get from grammy/telegraf.
-   * @returns A number in **string** format as the unique key (identifier).
-   * @description A function to generate a unique key for every user. You cound set it as any key you want (e.g group id)
+   * @returns A unique **string** key (identifier).
+   * @description A function to generate a unique key for every user. You could set it as any key you want (e.g. group id)
    * @see [Getting Started](https://github.com/Amir-Zouerami/rateLimiter#-how-to-use)
    */
   keyGenerator?: (ctx: C) => string | undefined;
+
+  /**
+   * @default "RATE_LIMITER"
+   * @description A string prefix that is getting added to the storage key after calling the `keyGenerator()`.
+   */
+  keyPrefix?: string | undefined;
 }
 
-export const defaultOptions = {
+export const defaultOptions: OptionsInterface<Context, RedisType> = {
```

Comment:
```suggestion
export const defaultOptions: Required<OptionsInterface<Context, RedisType>> = {
```

The `OptionsInterface` is the type for user options sent to rateLimiter by the consumer which is why all the fields are nullable (`?`). This code would cause type errors in rateLimiter.ts because the defaults cannot have undefined fields and must be present.

If you want to explicitly type this, you can use `Required` helper type.

Reply:
Okay, I've improved the typing for this.